File: talkos-service/src/main/java/ua/vn/talkos/service/UserService.java
package ua.vn.talkos.service;

import org.springframework.security.core.userdetails.UserDetailsService;
import ua.vn.talkos.dto.UserDto;
import ua.vn.talkos.entity.User;

import java.util.List;

/**
 * @author oleg.sukhov
 */
public interface UserService extends UserDetailsService {
    List<User> loadAll();
    User loadByLogin(String username);
    void save(UserDto userDto);
}

File: talkos-web/src/main/webapp/resources/js/app/user/user.service.js
angular.module('talkos')
    .service('UserService', ['$resource',
        function ($resource) {
            return {

                save: function(user) {
                    var userResource = $resource('/register', {'save': {method:'POST'}});
                    userResource.save(user);
                }
            };
        }]);
